client/client-multi/client-cpp/impl/log/LogCollector.cpp

```diff
@@ -36,17 +36,17 @@
 
 namespace kaa {
 
-LogCollector::LogCollector(IKaaChannelManagerPtr manager, IExecutorContext& executorContext, const KaaClientProperties& clientProperties)
+LogCollector::LogCollector(IKaaChannelManagerPtr manager, IKaaClientContext &context)
     : transport_(nullptr), channelManager_(manager), timeoutAccessPointId_(0),
       logUploadCheckTimer_("LogCollector logUploadCheckTimer"), scheduledUploadTimer_("LogCollector uploadTimer"),
-      timeoutTimer_("LogCollector timeoutTimer"), executorContext_(executorContext)
+      timeoutTimer_("LogCollector timeoutTimer"), context_(context)
 {
 #ifdef KAA_USE_SQLITE_LOG_STORAGE
-    storage_.reset(new SQLiteDBLogStorage(clientProperties.getLogsDatabaseFileName()));
+    storage_.reset(new SQLiteDBLogStorage(context_, context_.getProperties().getLogsDatabaseFileName()));
```

Comment:
There are a lot of cases when a user want to use one of default log storages but with other configuration (for example, different database file name). And after following changes he(she) won't be able to re-use these classes.

As a workaround we can add additional API call to retrieve a context from client, but it isn't a good idea.